### DISCUSSION TITLE
Make AnimationPlayer could ignore Engine.TimeScale

### DISCUSSION THREAD

**Discussion by Jamesika** [godotengine/godot-proposals#7706] [General Discussions]
(This is another guy escape from Unity : ))
use-case:
When it comes to make a Simulation Game, "Change TimeScale" is a common feature, such as The Sims series. When Engine.TimeScale changed, all AnimationPlayer is affected, However, the GUI Animation should keep 1x speed as usual. In Unity, there is a enum value to make Animator ignore TimeScale:
![image](https://github.com/godotengine/godot-proposals/assets/23502082/fee55d09-f3f5-4711-87da-ef39a4e2a3a6)

Maybe Godot should implement that. 
(Timer should also support such a feature)

**Comment by AThousandShips** [comment]
I'm not sure using the time scale as you suggest is the appropriate way to accomplish such, you can provide time scale to animation already and control the scale too

**Reply by Jamesika** [reply]
If I manually simulate a new custom TimeScale, there is a lot of work to do to fit that custom TimeScale, Particles/Timer/Animations/Physics/....
But if I use Engine.TimeScale, the only work is to make GUI seems normal.

**Reply by AThousandShips** [reply]
The support for such might be complex to implement, but you can manually change the scale of your animations that should stay fixed easily with [this property](https://docs.godotengine.org/en/stable/classes/class_animationplayer.html#class-animationplayer-property-speed-scale)

**Reply by Jamesika** [reply]
Yes, that's my current solution. I'm here to suggest implementing this feature in GodotEngine. (Because I found that Godot Nodes lack unscaled time support)

**Comment by AThousandShips** [comment]
The more general time scale control like Unity was already suggested and rejected [here](https://github.com/godotengine/godot-proposals/issues/2507)

See there for discussion, but this might be closed as well for the same reasons
